projects/common/src/time/time-duration.ts

```diff
@@ -26,6 +29,70 @@ export class TimeDuration {
     return `PT${this.toMillis() / 1000}S`;
   }
 
+  public toIso8601DurationStringMilitaryTime(): string {
+    const timeObject = this.toTimeObject();
+
+    let timeString = 'P';
+    if (timeObject.years > 0) {
+      timeString = timeString.concat(toString(timeObject.years), 'Y');
+    }
+    if (timeObject.months > 0) {
+      timeString = timeString.concat(toString(timeObject.months), 'M');
+    }
+    if (timeObject.days > 0) {
+      timeString = timeString.concat(toString(timeObject.days), 'D');
+    }
+    if (timeObject.hours > 0 || timeObject.minutes > 0 || timeObject.seconds > 0) {
+      timeString = timeString.concat('T');
+    }
+    if (timeObject.hours > 0) {
+      timeString = timeString.concat(toString(timeObject.hours), 'H');
+    }
+    if (timeObject.minutes > 0) {
+      timeString = timeString.concat(toString(timeObject.minutes), 'M');
+    }
+    if (timeObject.seconds > 0) {
+      timeString = timeString.concat(toString(timeObject.seconds), 'S');
+    }
+
+    return timeString;
+  }
+
+  private toTimeObject(): TimeObject {
+    return {
+      years: this.years,
+      months: this.months,
+      days: this.days,
+      hours: this.hours,
+      minutes: this.minutes,
+      seconds: this.seconds
+    };
+  }
+
+  public get years(): number {
+    return this.militaryTime.getUTCFullYear() - new Date(0).getUTCFullYear();
+  }
+
+  public get months(): number {
+    return this.militaryTime.getUTCMonth() - new Date(0).getUTCMonth();
```

Comment:
All of these things based on the epoch instant are statics, and would be more clearly just defined as such. `new Date(0).getUTCMonth()` is always going to be 0 for example. `new Date(0).getUTCDate()` will always be 1 (because confusingly, days are the one unit that's based)